nautobot/extras/tests/test_datasources.py

```diff
@@ -192,6 +192,12 @@ def empty_repo(self, path, url, *args, **kwargs):
         os.remove(os.path.join(path, "jobs", "__init__.py"))
         return mock.DEFAULT
 
+    def assert_repo_slug_valid_python_package_name(self):
+        git_repository = GitRepository.objects.create(
+            name="1 Very-Bad Git_____Repo Name (2)", remote_url="http://localhost/git.git"
+        )
+        self.assertEqual(git_repository.slug, "a1_very_bad_git_____repo_name_2")
```

Comment:
Where's the `a` at the beginning coming from?

Reply:
GraphQL identifiers (and Python module names) can't start with a digit, so the slugify function automatically prepends `a` in such a case to make it valid.